packages/mobile/src/fiatExchanges/FiatExchangeAmount.test.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+import { fireEvent, render } from 'react-native-testing-library'
+import { Provider } from 'react-redux'
+import { ExchangeRatePair } from 'src/exchange/reducer'
+import FiatExchangeAmount from 'src/fiatExchanges/FiatExchangeAmount'
+import { CURRENCY_ENUM } from 'src/geth/consts'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { createMockStore, getMockStackScreenProps } from 'test/utils'
+
+const exchangeRatePair: ExchangeRatePair = { goldMaker: '0.5', dollarMaker: '1' }
+
+const mockScreenProps = getMockStackScreenProps(Screens.FiatExchangeAmount, {
+  currency: CURRENCY_ENUM.DOLLAR,
+})
+
+describe('FiatExchangeAmount', () => {
+  const store = createMockStore({
+    stableToken: {
+      balance: '1000.00',
+    },
+    exchange: { exchangeRatePair },
+  })
+
+  it('renders correctly', () => {
+    const { toJSON } = render(
+      <Provider store={store}>
+        <FiatExchangeAmount {...mockScreenProps} />
+      </Provider>
+    )
+    expect(toJSON()).toMatchSnapshot()
+  })
+
+  it('validates the amount when cashing in', () => {
```

Comment:
Nit: this is a very comprehensive test but I think it would be better if we broke this out a little more into smaller tests that are verifying distinct code paths. It will be more code but should give us more granular detail if/when something fails